### ISSUE TITLE
SVM: Fix the HDRNAME keyword values generated for the headerlets

### ISSUE THREAD

**Issue by stscijgbot-hstdp** [spacetelescope/drizzlepac#1203]
_Issue [HLA-671](https://jira.stsci.edu/browse/HLA-671) was created on JIRA by [Michele De La Pena](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mdelapena):_

When creating the unique HDRNAME values for the headerlets during SVM processing, the SVM FLT/FLC filename is used as the base (minus the ".fits").   The base is then appended with the WCSNAME + "-hlet.fits".  However, the "t"or "c" is currently also being chopped off the "fl"of the base name.  File hst_9029_01_acs_wfc_f775w_j8ca01at_flt.fits had only "OPUS" and "IDC_0461802ej" WCS solutions in the original ipppssoot.  After SVM processing, the file now contains the WCS solutions seen in the box below.  The HDRNAMEs added by the SVM drizzling or downstream processing contain the string "j8ca01at_fl_IDC_" which should be "j8ca01at_flc_IDC_".  As an aside, the same headerlet has been added twice which is not harmful, but it does add to the complexity for user understanding.
```
                            WCSNAME     HDRNAME                                                                   
     IDC_0461802ej-FIT_SVM_GAIAeDR3     hst_9029_01_acs_wfc_f775w_j8ca01at_fl_IDC_0461802ej-FIT_SVM_GAIAeDR3-hlet.fits
                      IDC_0461802ej     hst_9029_01_acs_wfc_f775w_j8ca01at_flt_IDC_0461802ej-hlet.fits            
     IDC_0461802ej-FIT_SVM_GAIAeDR3     hst_9029_01_acs_wfc_f775w_j8ca01at_fl_IDC_0461802ej-FIT_SVM_GAIAeDR3-hlet.fits                      
KEY WCSNAME
O   OPUS
A   IDC_0461802ej
B   IDC_0461802ej-FIT_SVM_GAIAeDR3
    IDC_0461802ej```

**Comment by stscijgbot-hstdp** [comment]
_Comment by [Warren Hack](https://jira.stsci.edu/secure/ViewProfile.jspa?name=hack) on [JIRA](https://jira.stsci.edu/browse/HLA-671?focusedCommentId=565733#comment-565733):_

Need to confirm whether or not [Drizzlepac PR #&#x2060;1202](https://github.com/spacetelescope/drizzlepac/pull/1202) addressed this problem.

**Comment by stsci-hack** [comment]
Closed by #1202.

**Comment by stscijgbot-hstdp** [comment]
_Comment by [Warren Hack](https://jira.stsci.edu/secure/ViewProfile.jspa?name=hack) on [JIRA](https://jira.stsci.edu/browse/HLA-671?focusedCommentId=570637#comment-570637):_

It turns out that the 'flt' or 'flc' is being turned into 'fl' in the _haputils.align_utils.update_image_wcs_info()_ function when the HDRNAME gets defined.  This comes as a result of the use of the 'rstrip()' function when applied to a full filename.

> a
>  Out[7]: 'hst_13158_03_acs_sbc_f122m_jc6e03uf_flt.fits'
>  a.rstrip('.fits')
>  Out[8]: 'hst_13158_03_acs_sbc_f122m_jc6e03uf_fl'
>  a.rstrip('fits')
>  Out[9]: 'hst_13158_03_acs_sbc_f122m_jc6e03uf_flt.'

This finally gets fixed in [Drizzlepac PR #&#x2060;1265](https://github.com/spacetelescope/drizzlepac/pull/1265).   This results in a unique set of WCSNAME/HDRNAME solutions in the output FLT headers after SVM processing:

 

> flts[0]
>  Out[13]: 'hst_13158_03_acs_sbc_f122m_jc6e03uf_flt.fits'
> 
> fits.getval(flts[0], 'wcsname*', ext=1)
>  Out[14]: 
>  WCSNAMEO= 'OPUS ' / Coordinate system title 
>  WCSNAMEA= 'IDC_v6m2008mj' / Coordinate system title 
>  WCSNAMEB= 'IDC_v6m2008mj-HSC30' / Coordinate system title 
>  WCSNAMEC= 'IDC_v6m2008mj-FIT_IMG_GAIAeDR3' / Coordinate system title 
>  WCSNAMED= 'IDC_v6m2008mj-FIT_SVM_GAIAeDR3' / Coordinate system title 
>  WCSNAME = 'IDC_v6m2008mj-GSC240' / Coordinate system title 
> 
> fits.getval(flts[0], 'hdrname*', ext=1)
>  Out[15]: 
>  HDRNAMEA= 'jc6e03ufq_fl_IDC_v6m2008mj-FIT_IMG_GAIAeDR3-hlet.fits' 
>  HDRNAMEB= 'jc6e03ufq_flt_IDC_v6m2008mj-HSC30' 
>  HDRNAMEC= 'jc6e03ufq_fl_IDC_v6m2008mj-FIT_IMG_GAIAeDR3-hlet.fits' 
>  HDRNAME = 'jc6e03ufq_flt_IDC_v6m2008mj-GSC240' 
>  HDRNAMED= 'hst_13158_03_acs_sbc_f122m_jc6e03uf_flt_IDC_v6m2008mj-FIT_SVM_GAIAe&'
>  CONTINUE 'DR3-hlet.fits'
